components/governance/org.wso2.carbon.governance.api/src/main/java/org/wso2/carbon/governance/api/util/GovernanceUtils.java

```diff
@@ -1934,10 +1934,27 @@ public static void setDefaultLifeCycle(String path, String aspect, Registry regi
 
             List<GovernanceArtifact> propertySearchResults = performAttributeSearch(fields, registry);
 
-            Set<GovernanceArtifact> set  = new TreeSet<>(new Comparator<GovernanceArtifact>() {
-                public int compare(GovernanceArtifact artifact1, GovernanceArtifact artifact2)
-                {
-                    return artifact1.getId().compareTo(artifact2.getId()) ;
+            Set<GovernanceArtifact> set = new TreeSet<>(new Comparator<GovernanceArtifact>() {
+                public int compare(GovernanceArtifact artifact1, GovernanceArtifact artifact2) {
+                    PaginationContext paginationContext = PaginationContext.getInstance();
+                    if(paginationContext != null) {
+                        String sortBy = paginationContext.getSortBy();
+                        try {
+                            switch (paginationContext.getSortOrder()) {
+                                case "ASC":
+                                    return artifact1.getAttribute(sortBy).compareTo(artifact2.getAttribute(sortBy));
```

Comment:
Chance of a null pointer exception. Because attribute can be a null value. Hence compareto method of a nulll value can cause NPE